pkg/controller/llmisvc/router.go

```diff
@@ -0,0 +1,195 @@
+)
+
+func (r *LLMInferenceServiceReconciler) reconcileRouter(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService) error {
+	logger := log.FromContext(ctx)
+
+	logger.Info("Reconciling Router")
+
+	if err := r.reconcileScheduler(ctx, llmSvc); err != nil {
+		llmSvc.MarkRouterNotReady("SchedulerReconcileError", "Failed to reconcile scheduler: %v", err.Error())
+		return fmt.Errorf("failed to reconcile workload: %w", err)
+	}
+
+	if err := r.reconcileGateway(ctx, llmSvc); err != nil {
+		llmSvc.MarkRouterNotReady("GatewayReconcileError", "Failed to reconcile gateway: %v", err.Error())
+		return err
+	}
+
+	// TODO: reconcile networkingv1.Ingress
+
+	if err := r.reconcileHTTPRoutes(ctx, llmSvc); err != nil {
+		llmSvc.MarkRouterNotReady("HTTPRouteReconcileError", "Failed to reconcile HTTPRoute: %v", err.Error())
+		return err
+	}
+
+	llmSvc.MarkRouterReady()
+
+	return nil
+}
+
+func (r *LLMInferenceServiceReconciler) reconcileGateway(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService) error {
+	logger := log.FromContext(ctx)
+	logger.Info("Reconciling Gateway... coming soon")
+
+	return nil
+}
+
+func (r *LLMInferenceServiceReconciler) reconcileHTTPRoutes(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService) error {
+	logger := log.FromContext(ctx)
+	logger.Info("Reconciling HTTPRoute")
+
+	expectedHTTPRoute := r.expectedHTTPRoute(llmSvc)
+
+	if llmSvc.Spec.Router == nil || llmSvc.Spec.Router.Route == nil || llmSvc.Spec.Router.Route.HTTP == nil {
+		return r.deleteHTTPRoute(ctx, llmSvc, expectedHTTPRoute)
+	}
+
+	route := llmSvc.Spec.Router.Route
+
+	if route.HTTP.HasRefs() {
+		// We should use provided HTTPRoutes, clean ours up if they exist
+		return r.deleteHTTPRoute(ctx, llmSvc, expectedHTTPRoute)
+	}
+
+	if route.IsManaged() || route.HTTP.HasSpec() {
+		return r.reconcileHTTPRoute(ctx, llmSvc, expectedHTTPRoute)
+	}
+
+	return nil
+}
+
+func (r *LLMInferenceServiceReconciler) expectedHTTPRoute(llmSvc *v1alpha1.LLMInferenceService) *gatewayapi.HTTPRoute {
+	httpRoute := &gatewayapi.HTTPRoute{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kmeta.ChildName(llmSvc.GetName(), "-kserve-route"),
+			Namespace: llmSvc.GetNamespace(),
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(llmSvc, v1alpha1.LLMInferenceServiceGVK),
+			},
+			Labels: RouterLabels(llmSvc),
+		},
+	}
+
+	if llmSvc.Spec.Router != nil && llmSvc.Spec.Router.Route != nil && llmSvc.Spec.Router.Route.HTTP.Spec != nil {
+		httpRoute.Spec = *llmSvc.Spec.Router.Route.HTTP.Spec.DeepCopy()
+	}
+
+	if llmSvc.Spec.Router != nil && llmSvc.Spec.Router.Gateway != nil {
+		for _, ref := range llmSvc.Spec.Router.Gateway.Refs {
+			httpRoute.Spec.CommonRouteSpec.ParentRefs = append(httpRoute.Spec.CommonRouteSpec.ParentRefs, gatewayapi.ParentReference{
+				// TODO(api): With this structure we are missing the ability to narrow a section of targeted gateway by the route we are creating
+				// missing SectionName and Port
```

Comment:
This is what we disccussed @pierDipi - not sure we want to open an issue for the future enhancement. WDYT?

Reply:
I will take care of it.